packaging/installer/functions.sh

```diff
@@ -374,7 +374,13 @@ install_netdata_service() {
 
             SYSTEMD_DIRECTORY=""
 
-            if [ -d "/lib/systemd/system" ]; then
+            if [ -f "/etc/os-release" ]; then
+                source "/etc/os-release" || return 1
+            fi
+
+            if [ x"$ID" = x"coreos" ]; then
+                SYSTEMD_DIRECTORY="/etc/systemd/system"
```

Comment:
While this does work, I'm kind of the opinion that this doesn't completely fix #5558. We should be checking writability of each of the places we're trying to put the unit file, and picking the first writable one in the list (with `/etc/systemd/system` being at the end of the list), as there are other reasons that `/lib/systemd/system` and `/usr` might be read-only besides running on CoreOS.

Reply:
Ahhh. Sorry I commented on your earlier comment too soon! Let's not have a discussion in both places! :)

So... In actuality I did try it this way at first only to realise that the level of code/effort required was vastly more complicated than just "Are we on CoreOS? Yes? Then write here."

I do agree with you; but I _feel_ this will become more apparently as we run into more edge-cases and esoteric SystemD configurations on other systems. If however the variations are small then the extra engineering effort/complexity won't be worth it anyway.

Reply:
> So... In actuality I did try it this way at first only to realise that the level of code/effort required was vastly more complicated than just "Are we on CoreOS? Yes? Then write here."

Could you elabolate a bit more regarding why it is complicated? I am thinking it would be a matter of testing for writeability with something like '[ -w /path/to/dir ]' but I guess I am missing something

Reply:
TIL: `[ -w /path/to/dir ]` <-- 🤣

Given @Ferroin's comment below I'll look into this again. Perhaps doing this is easier than I first though :)